### PR TITLE
Add KUBECONFIG environment variable handling to kubernetes adapter.

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR primarily adds handling to kubernetes adapter to take the
value for kubeconfigPath from mixer config as well as from environmental
config (via the KUBECONFIG env var).

NOTE: this configuration was previously available with via
KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT env vars. This PR
consolidates that into the Istio standard KUBECONFIG env var to
harmonize deployment across componenets.

Tested with out-of-the-box config via:

```bash
$ KUBECONFIG=~/.kube/config bazel-bin/cmd/server/mixs server --configStoreURL=fs://$(pwd)/testdata/configroot --logtostderr -v=5
```

and several client calls, as follows:

```bash
$ bazel-bin/cmd/client/mixc report -v=45 -a target.service=foo.default.svc.cluster.local --string_attributes source.uid=kubernetes://productpage-v1-2087264475-bl9f2.default -m localhost:9091 --trace
Report RPC returned OK
Attribute              Type               Value
source.labels          map[string]string  map[version:v1 app:productpage pod-template-hash:2087264475]
source.name            string             productpage-v1-2087264475-bl9f2
source.namespace       string             default
source.service         string             productpage.default.svc.cluster.local
source.serviceAccount  string             default
```

This PR also fixes the previously observed issue with sync.Once and initialization.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/729)

### CODE REVIEW

File: adapter/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -184,23 +193,30 @@ func (*builder) ValidateConfig(c adapter.Config) (ce *adapter.ConfigErrors) {
...
+	b.Lock()
+	defer b.Unlock()
+	if b.needsCacheInit {
 		refresh := paramsProto.CacheRefreshDuration
-		controller, err := b.newCacheControllerFn(paramsProto.KubeconfigPath, refresh, env)
+		path, exists := os.LookupEnv("KUBECONFIG")
+		if !exists {
+			path = paramsProto.KubeconfigPath
+		}
+		controller, err := b.newCacheControllerFn(path, refresh, env)
```
Comment (mandarjog) [inline adapter/kubernetes/kubernetes.go]: Should add debug log, re: which kubconfig `path` was finally used.

File: adapter/kubernetes/kubernetes_test.go [hunk 2]
```diff
@@ -145,6 +147,39 @@ func TestBuilder_BuildAttributesGenerator(t *testing.T) {
...
+		name    string
+		testFn  controllerFactoryFn
+		conf    adapter.Config
+		wantErr bool
+	}{
+		{"success", fakePodCache, conf, false},
+	}
+
+	wantPath := "/want/kubeconfig"
+	if err := os.Setenv("KUBECONFIG", wantPath); err != nil {
```
Comment (mandarjog) [inline adapter/kubernetes/kubernetes_test.go]: test for what take precedence  between env var and proto?

File: adapter/kubernetes/kubernetes.go (hunk 1 above)
Comment (douglas-reid) [inline adapter/kubernetes/kubernetes.go]: this happens in the first line of `newCacheControllerFn`:

`env.Logger().Infof("getting kubeconfig from: %#v", kubeconfigPath)`

File: adapter/kubernetes/kubernetes_test.go (hunk 2 above)
Comment (douglas-reid) [inline adapter/kubernetes/kubernetes_test.go]: done.

**Comment by douglas-reid** [comment]
Thanks for the review. I've updated the test to match comments. PTAL.

**Pull request merged**